### PR TITLE
Bugfix/lattice 2265 clear orphaned linking data

### PR DESCRIPTION
**Pull request by you-leee**

### CODE REVIEW

File: src/main/kotlin/com/openlattice/mechanic/MechanicCli.kt [hunk 1]
```diff
@@ -98,15 +94,32 @@ class MechanicCli {
...
+                .argName("name")
+                .desc("Run regeneration on the system. ")
+                .valueSeparator(',')
+                .build()
+
+        private val retireOption = Option.builder()
+                .longOpt(RETIRE)
+                .hasArgs()
+                .argName("name")
+                .desc("Run regeneration on the system. ")
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/mechanic/MechanicCli.kt]: I think this is meant to be something else?

File: src/main/kotlin/com/openlattice/mechanic/regenerate/Regeneration.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * You can contact the owner of the copyright at support@openlattice.com
+ *
+ *
+ */
+package com.openlattice.mechanic.regenerate
+
+interface Regeneration {
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/mechanic/regenerate/Regeneration.kt]: Shouldn't this also be a versioned task? Or is there some alternate ordering to these?

File: src/main/kotlin/com/openlattice/mechanic/regenerate/RegenerateIds.kt [hunk 3]
```diff
@@ -60,22 +55,18 @@ import java.util.function.Supplier
...
-    private val entityTypes = etms.loadAllKeys()?.map { it to etms.load(it) }?.toMap() ?: mapOf()
-    private val propertyTypes = ptms.loadAllKeys()?.map { it to ptms.load(it) }?.toMap() ?: mapOf()
+        private val principalTrees: PrincipalTreesMapstore
+) : Regeneration {
     private val ranges = idGen.loadAllKeys()?.map { it to idGen.load(it) }?.toMap()?.toMutableMap() ?: mutableMapOf()
     private val rangeIndex = AtomicLong()
     private val r = Random()
 
+    override fun regenerate(): Boolean {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/mechanic/regenerate/RegenerateIds.kt]: Looks like this is all unused, presumably b/c this isn't implemented yet

File: src/main/kotlin/com/openlattice/mechanic/Mechanic.kt [hunk 4]
```diff
@@ -194,6 +204,19 @@ class Mechanic {
         }
     }
 
+    fun regenerate(regenNames: Set<String>) {
+        val regens = context.getBeansOfType(Regeneration::class.java)
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/mechanic/Mechanic.kt]: Can replace this and `doUpgrade` with a generic method that takes a class and some logging templates

File: src/main/kotlin/com/openlattice/mechanic/integrity/OrphanedLinkingPropertyData.kt [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        val count = toolbox.hds.connection.use { conn ->
+            conn.createStatement().use {
+                it.executeUpdate(tombStoneSql)
+            }
+        }
+        logger.info("Cleared $count orphaned linking entries in ${DATA.name} table in ${sw.elapsed(TimeUnit.MILLISECONDS)} ms.")
+    }
+
+    private fun deleteOrphanedLinkingProperties() {
+        val sw = Stopwatch.createStarted()
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/mechanic/integrity/OrphanedLinkingPropertyData.kt:55]: This and the above can also be one generic method

**Review by makosblade — COMMENTED** [review]
Some comments, haven't fully reviewed `OrphanedLinkingPropertyData.kt` fully yet

File: src/main/kotlin/com/openlattice/mechanic/regenerate/Regeneration.kt (hunk 2 above)
Comment (you-leee) [reply src/main/kotlin/com/openlattice/mechanic/regenerate/Regeneration.kt]: Probably yes.. at least from `RegenerateIds` it seems so

File: src/main/kotlin/com/openlattice/mechanic/regenerate/RegenerateIds.kt (hunk 3 above)
Comment (you-leee) [reply src/main/kotlin/com/openlattice/mechanic/regenerate/RegenerateIds.kt]: Lot of the classes seem to be unused, but I wasn't sure if I should delete it or not. If we all agree, I can just delete it

Comment (makosblade) [reply src/main/kotlin/com/openlattice/mechanic/regenerate/RegenerateIds.kt]: I vote delete as I am a big fan of deleting as much code as possible 👍

File: src/main/kotlin/com/openlattice/mechanic/Mechanic.kt (hunk 4 above)
Comment (you-leee) [reply src/main/kotlin/com/openlattice/mechanic/Mechanic.kt]: Unless we specify a common interface for them, I wouldn't do this in a generic method

Comment (makosblade) [reply src/main/kotlin/com/openlattice/mechanic/Mechanic.kt]: 👍 no worries, this is fine for now

**Review by pckeyes — APPROVED** [review]

**Review by dbt — APPROVED** [review]

File: src/main/kotlin/com/openlattice/mechanic/regenerate/RegenerateIds.kt (hunk 3 above)
Comment (makosblade) [reply src/main/kotlin/com/openlattice/mechanic/regenerate/RegenerateIds.kt]: Looks like 4 votes for delete! 🎉

**Review by makosblade — APPROVED** [review]

File: src/main/kotlin/com/openlattice/mechanic/regenerate/RegenerateIds.kt (hunk 3 above)
Comment (you-leee) [reply src/main/kotlin/com/openlattice/mechanic/regenerate/RegenerateIds.kt]: 🔪

Comment (geekbeast) [reply src/main/kotlin/com/openlattice/mechanic/regenerate/RegenerateIds.kt]: O.O It's fine. I think there is a better way of doing what this was originally intended to do anyway.

**Pull request merged**